### Problem 1
1'den 1000'e kadar olan sayılardan mükemmel sayı olanları ekrana yazdırın. Bunun için bir sayının mükeı dmmel olup olmadığınönen bir tane fonksiyon yazın.
*Bir sayının bölenlerinin toplamı kendine eşitse bu sayı mükemmel bir sayıdır. Örnek olarak 6 mükemmel bir sayıdır (1 + 2 + 3 = 6).*

In [1]:
def mükemmel_sayı(sayı):
    
    toplam = 0
    
    for i in range(1, sayı):
        if sayı % i == 0:
            toplam += i
    
    return toplam == sayı

sonuc = []

for i in range(1,1001):
    if mükemmel_sayı(i):
        sonuc.append(i)
        
print(sonuc)

[6, 28, 496]


In [5]:
sonuc = []

for tüm_değerler in range(1, 1001):

    toplam = 0

    for bölenler in range(1, tüm_değerler):
        if tüm_değerler % bölenler == 0:
            toplam += bölenler

    if toplam == tüm_değerler:
        sonuc.append(tüm_değerler)

print(sonuc)

[6, 28, 496]


In [33]:
sonuc =[]

def sayısay(sayı) :
    toplam = 0

    for i in range(len(str(sayı))):
        toplam += int(str(sayı)[i])       
    return 2 * toplam == sayı 

for i in range(1, 101):
    if sayısay(i) :
        sonuc.append(i)
print(sonuc)             

[18]


In [7]:
sonuc = []

for tüm_değerler in range(1, 101):

    toplam = 0

    for i in range(len(str(tüm_değerler))):
        toplam += int(str(tüm_değerler)[i])

    if 2*toplam == tüm_değerler :
        sonuc.append(tüm_değerler)

print(sonuc)

[18]


In [40]:
sonuc = []
def armstrong_number(number) :
    summ = 0
    for i in range(len(str(number))):
        summ += int(str(number)[i]) **3
    return summ == number

for i in range(1,1001) :
    if armstrong_number(i):
        sonuc.append(i)

print(sonuc) 

[1, 153, 370, 371, 407]


In [37]:
sonuc =[]

def armstrong_number(number) :
    summ = 0
    for i in range(len(str(number))):
        summ += int(str(number)[i]) **3
    return summ == number

for number in range(1, 1001):
    if armstrong_number(number) :
        sonuc.append(number)

print(sonuc)

[1, 153, 370, 371, 407]


In [14]:
num = int(input("bir sayı gir"))
summ = 0
for i in range(len(str(num))):
        summ += int(str(num)[i]) **3

if summ == num :
    print(num, "is a ARMSTRONG number. ")

370 is a ARMSTRONG number. 


### Problem 2
Kullanıcıdan 2 tane sayı alarak bu sayıların en büyük ortak bölenini (EBOB) dönen bir tane fonksiyon yazın. 
Problem için şu siteye bakabilirsiniz;
http://www.matematikciler.com/6-sinif/matematik-konu-anlatimlari/1020-en-kucuk-ortak-kat-ve-en-buyuk-ortak-bolen-ebob-ekok
matematikciler.commatematikciler.com
8. Sınıf EBOB - EKOK Konu Anlatımı | matematikciler.com
8. sınıf ebob - ekok konu anlatımı. Bu konuda en büyük ortak bölen, en küçük ortak kat, ebob-ekok bulma, ebob-ekok problemleri anlatılmaktadır. (45 kB)

In [18]:
def ebob_bul(sayi1, sayi2) :
    if(sayi1>sayi2) :
        kücük = sayi2
    else:
        kücük = sayi1
    
    a = kücük

    while a>0 :
        if sayi1 % a == 0 and sayi2 % a == 0 :
            return a 
        a -=1
    return 1

while True :
    sayi1 = int(input("sayi1:"))
    sayi2 = int(input("sayi2:"))
    print(f"{sayi1} ve {sayi2}'in EBOB'u = {ebob_bul(sayi1, sayi2)}")
    break

100 ve 250'in EBOB'u = 50


Problem 3

Kullanıcıdan 2 tane sayı alarak bu sayıların en küçük ortak katlarını (EKOK) dönen bir tane fonksiyon yazın.
Problem için şu siteye bakabilirsiniz;
http://www.matematikciler.com/6-sinif/matematik-konu-anlatimlari/1020-en-kucuk-ortak-kat-ve-en-buyuk-ortak-bolen-ebob-ekok (bearbeitet) 

In [24]:
def ekok_bul(sayi1, sayi2) :
    if(sayi1>sayi2) :
        büyük = sayi1
    else:
        büyük = sayi2
    
    a = büyük

    while a < sayi1 * sayi2 :
        if a % sayi1 == 0 and a % sayi2 == 0 :
            return a 
        a +=1
    return sayi1 * sayi2 

while True :
    sayi1 = int(input("sayi1:"))
    sayi2 = int(input("sayi2:"))
    print(f"{sayi1} ve {sayi2}'in EKOK'u = {ekok_bul(sayi1, sayi2)}")
    break

9 ve 12'in EKOK'u = 36


### Problem 4

Kullanıcıdan 2 basamaklı bir sayı alın ve bu sayının okunuşunu bulan bir fonksiyon yazın.
*Örnek: 97 ---------> Doksan Yedi*

In [45]:
def readanumber(number) :
    a = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    b = ["", "ten", "twenty", "thirty","fourty", "fifty", "sixty", "seventy", "eighty", "ninety"]
    c = ["", "onehundred", "twohundred", "threehundred", "fourhundred", "fivehundred","sixhundred", "sevenhundred", "eighthundred","ninehundred"]
    d = ["", "onethausand", "twothausand", "threethausand", "fourthausand", "fivethausand", "sixthausand", "seventhausand", "eightthausand", "ninethausand"]

    return str(d[int((str(number))[0])]) +  str(c[int((str(number))[1])]) +  str(b[int((str(number))[2])]) +  str(a[int((str(number))[3])])


In [47]:
print(readanumber(9786))

ninethausandsevenhundredeightysix


Problem 5

1'den 100'e kadar olan sayılardan pisagor üçgeni oluşturanları ekrana yazdıran bir fonksiyon yazın.(a <= 100,b <= 100)

In [28]:
nlist = []
for a in range(1,101) :
    for b in range(1,101) :
        for c in range(1,101):
        
            if a**2 == b**2 + c**2 or b**2 == a**2 + c**2 or c**2 == a**2 + b**2 :
                borders={a,b,c}
                if borders not in nlist :
                    nlist.append(borders)
            

print(len(nlist), nlist, sep="\n")

52
[{3, 4, 5}, {13, 12, 5}, {8, 10, 6}, {24, 25, 7}, {8, 17, 15}, {9, 12, 15}, {40, 9, 41}, {24, 10, 26}, {11, 60, 61}, {16, 12, 20}, {35, 12, 37}, {85, 84, 13}, {48, 50, 14}, {25, 20, 15}, {36, 39, 15}, {16, 34, 30}, {16, 65, 63}, {24, 18, 30}, {80, 18, 82}, {29, 20, 21}, {48, 20, 52}, {35, 28, 21}, {72, 75, 21}, {24, 32, 40}, {24, 51, 45}, {24, 74, 70}, {65, 25, 60}, {27, 36, 45}, {53, 28, 45}, {96, 100, 28}, {40, 50, 30}, {72, 78, 30}, {32, 60, 68}, {33, 44, 55}, {56, 33, 65}, {91, 35, 84}, {48, 36, 60}, {85, 36, 77}, {65, 52, 39}, {80, 89, 39}, {40, 42, 58}, {40, 75, 85}, {56, 42, 70}, {75, 60, 45}, {48, 73, 55}, {48, 80, 64}, {51, 68, 85}, {72, 90, 54}, {57, 76, 95}, {60, 87, 63}, {80, 100, 60}, {72, 65, 97}]


In [30]:
### EDEBALI
pythagoras = []
for i in range(1,101):
    for j in range(1,101):
            for k in range(1,101):
                if i+j >k > abs(i-j) :
                    if j**2 + k**2 == i**2:
                        borders = {i, j, k}
                        if borders not in pythagoras:
                            pythagoras.append(borders)

print(f'Under these conditions, there are {len(pythagoras)} unique Pythagorean triangles : {pythagoras}')

Under these conditions, there are 52 unique Pythagorean triangles : [{3, 4, 5}, {8, 10, 6}, {5, 12, 13}, {9, 12, 15}, {8, 17, 15}, {16, 20, 12}, {24, 25, 7}, {25, 20, 15}, {24, 26, 10}, {21, 20, 29}, {24, 18, 30}, {16, 34, 30}, {35, 28, 21}, {35, 12, 37}, {15, 36, 39}, {40, 24, 32}, {40, 41, 9}, {27, 36, 45}, {48, 50, 14}, {40, 50, 30}, {24, 51, 45}, {48, 52, 20}, {45, 28, 53}, {33, 44, 55}, {40, 58, 42}, {48, 60, 36}, {11, 60, 61}, {16, 65, 63}, {65, 60, 25}, {56, 65, 33}, {65, 52, 39}, {32, 68, 60}, {56, 42, 70}, {48, 73, 55}, {24, 74, 70}, {72, 75, 21}, {75, 60, 45}, {72, 78, 30}, {80, 48, 64}, {80, 82, 18}, {13, 84, 85}, {77, 36, 85}, {40, 75, 85}, {51, 68, 85}, {60, 63, 87}, {80, 89, 39}, {72, 90, 54}, {35, 91, 84}, {57, 76, 95}, {65, 97, 72}, {96, 100, 28}, {80, 100, 60}]


2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

In [19]:
num = 2520
while True:
    count = 1
    for i in range(11,20):
        if not num % i :
            count += 1
    if count == 10:
        print(num)
        break
    num += 20

232792560


In [2]:
num = 2520
while True:
    count = 1
    for i in range(11,20):
        if num % i :
            break
        else : 
            count += 1
    if count == 10:
        print(num)
        break
    num += 20

232792560


1. QUESTİON
Task : Let's say you left a message before that prints a password you need. To see the password you left, you need to enter your name and the program should recognize you.Write a program that 
Takes the first name from the user and compares it to yours,
Then if the name the user entered is the same as yours, print out such as : "Hello, Joseph! The password is : @12",
If the name the user entered is not the same as yours, print out such as : "Hello, Amina! See you later."

In [50]:
name = "hans"
password = "2$1%54&635"

input_name = input("Enter a name : ")

if input_name == name :
    print(f"Hello, {name}! The password is : {password} ")
else :
    print(f"Hello, {input_name.title()}! See you later.")

Hello, Ahmet! See you later.


2. QUESTİON
Task:
Find out if a given number is an "Armstrong Number".
An n-digit number that is the sum of the nth powers of its digits is called an n-Armstrong number. Examples :
371 = 33 + 73 + 13;
9474 = 94 + 44 + 74 + 44;
93084 = 95 + 35 + 05 + 85 + 45.
Write a Python program that;
takes a positive integer number from the user,
checks the entered number if it is Armstrong,
consider the negative, float and any entries other than numeric values then display a warning message to the user.

In [9]:
while True :
    number = input("Enter a positive number : ")
    power = len(number)
    summ = 0

    if not number.isdigit() :
        print(number, "is not a valid entry! Enter a valid input.")

    elif int(number) >= 0 :

        for i in number :
            summ += int(i) ** power

        if summ == int(number) :
            print(number, "is an armstrong number.")
            break
        else :
            print(number, " is not a armstrong number. ")
            break 

371 is an armstrong number.


In [51]:
armstrong_numbers = []
def armstrong_number(number) :
    summ = 0
    for i in range(len(str(number))):
        summ += int(str(number)[i]) **3
    return summ == number

for i in range(1,1001) :
    if armstrong_number(i):
        armstrong_numbers.append(i)

print(armstrong_numbers) 

[1, 153, 370, 371, 407]
